### PR TITLE
numa_basic: Check node free memory before test

### PR DESCRIPTION
**Pull request by huangyum**
Check numa node free memory before running test

ID: 1503402
Signed-off-by: Yumei Huang <yuhuang@redhat.com>

### CODE REVIEW

**Comment by huangyum** [comment]
Test results:
When numa node has enough free memory, case passes.
 (1/1) smp_1.4096m.repeat1.run_test.Host_RHEL.m7.u5.qcow2.virtio_scsi.up.virtio_net.RHEL.7.5.x86_64.io-github-autotest-qemu.numa.numa_basic: PASS (291.68 s)

Otherwise, case skips.
 (1/1) smp_1.7649m.repeat1.run_test.Host_RHEL.m7.u5.qcow2.virtio_scsi.up.virtio_net.RHEL.7.5.x86_64.io-github-autotest-qemu.numa.numa_basic: SKIP

File: qemu/tests/numa_stress.py [hunk 1]
```diff
@@ -70,8 +71,8 @@ def run(test, params, env):
     """
     host_numa_node = utils_misc.NumaInfo()
     if len(host_numa_node.online_nodes) < 2:
-        raise error.TestNAError("Host only has one NUMA node, "
-                                "skipping test...")
+        raise exceptions.TestSkipError("Host only has one NUMA node, "
```
Comment (vivianQizhu) [inline qemu/tests/numa_stress.py]: We now use test.skip()

File: qemu/tests/numa_stress.py [hunk 2]
```diff
@@ -114,39 +115,39 @@ def run(test, params, env):
             os.remove(memory_file)
         utils_memory.drop_caches()
         if utils_memory.freememtotal() < tmpfs_size:
-            raise error.TestError("Don't have enough memory to execute this "
-                                  "test after %s round" % test_round)
-        error.context("Executing stress test round: %s" % test_round,
-                      logging.info)
+            raise exceptions.TestError("Don't have enough memory to execute this "
```
Comment (vivianQizhu) [inline qemu/tests/numa_stress.py]: test.error()

File: qemu/tests/numa_stress.py [hunk 3]
```diff
@@ -2,13 +2,14 @@
 import re
 import logging
 
-from autotest.client.shared import error
 from autotest.client import utils
+from avocado.core import exceptions
```
Comment (vivianQizhu) [inline qemu/tests/numa_stress.py]: Unnecessary

**Comment by vivianQizhu** [comment]
@huangyum Please use test module instead of exceptions for all process, thanks.

File: qemu/tests/numa_stress.py (hunk 1 above)
Comment (balamuruhans) [reply qemu/tests/numa_stress.py]: It is test.cancel()

**Comment by huangyum** [comment]
@vivianQizhu @balamuruhans Updated, please help review again, thanks!

File: qemu/tests/numa_basic.py [hunk 4]
```diff
@@ -23,24 +22,27 @@ def run(test, params, env):
...
     node_list = host_numa_node.online_nodes
     for node_id in node_list:
-        error.base_context("Bind qemu process to numa node %s" % node_id,
-                           logging.info)
+        error_context.base_context("Bind qemu process to numa node %s" % node_id,
+                                   logging.info)
         vm = "vm_bind_to_%s" % node_id
         params['qemu_command_prefix'] = "numactl --cpunodebind=%s" % node_id
         utils_memory.drop_caches()
+        node_MemFree = host_numa_node.read_from_node_meminfo(node_id, "MemFree")
```
Comment (balamuruhans) [inline qemu/tests/numa_basic.py]: This returns str, let us typecast to int
`node_MemFree = int(host_numa_node.read_from_node_meminfo(node_id, "MemFree"))`

File: qemu/tests/numa_stress.py [hunk 5]
```diff
@@ -114,39 +113,37 @@ def run(test, params, env):
...
         numa_dd_cmd = "numactl -m %s %s" % (numa_node_malloc, dd_cmd)
-        error.context("Try to allocate memory in node %s" % numa_node_malloc,
-                      logging.info)
+        error_context.context("Try to allocate memory in node %s" % numa_node_malloc,
+                              logging.info)
         try:
             utils_misc.mount("none", tmpfs_path, "tmpfs", perm=mount_fs_size)
             funcatexit.register(env, params.get("type"), utils_misc.umount,
-                                "none", tmpfs_path, "tmpfs")
+                                None, tmpfs_path, "tmpfs")
```
Comment (balamuruhans) [inline qemu/tests/numa_stress.py]: It is better to have a separate commit for this fix.

**Comment by balamuruhans** [comment]
Thanks @huangyum for the patch, @vivianQizhu for reviewing. Let us merge this PR.

**Pull request merged**